src/surge-xt/gui/overlays/FilterAnalysis.cpp

```diff
@@ -401,5 +440,50 @@ void FilterAnalysis::resized()
     catchUpStore = evaluator->outboundUpdates - 1; // because we need to rebuild the path
 }
 
+void FilterAnalysis::mouseDrag(const juce::MouseEvent &event) {
+    if (event.mods.isLeftButtonDown())
+    {
+        auto &ss = editor->getPatch().scene[editor->current_scene];
+        auto &fs = ss.filterunit[whichFilter];
+
+        auto lb = getLocalBounds().transformedBy(getTransform().inverted());
+        auto dRect = lb.withTrimmedTop(15).reduced(4);
+        auto width = dRect.getWidth();
+        auto height = dRect.getHeight();
+        juce::Point<int> mousePoint =
+            event
+                .getPosition()
+                .transformedBy(
+                    juce::AffineTransform()
+                                .translated(dRect.getX(), dRect.getY())
+                                .inverted()
+                );
+
+        auto xNorm = mousePoint.x / (float)width;
+        auto freq = std::pow(HIGHER_FREQ_IN_GRAPH / LOWER_FREQ_IN_GRAPH, xNorm) * LOWER_FREQ_IN_GRAPH;
+        static constexpr auto lowerLimit = -60.f;
+        static constexpr auto upperLimit = 70.f;
+        float cutoff = juce::jlimit(lowerLimit, upperLimit, 12.0f * std::log2(freq / 440.0f));
+        float resonance = juce::jlimit(0.f,1.f, (height - mousePoint.y) / (float)height);
+
+        float f =  juce::jmap(cutoff, lowerLimit, upperLimit, 0.0f, 1.0f );
+        fs.cutoff.val.f = cutoff;
+        fs.resonance.val.f = resonance;
+
+        jassert(editor->filterCutoffSlider[whichFilter] != nullptr); //  filter.cutoff_1 or filter.cutoff_2 is not assigned in SurgeGuiEditor.cpp
+        jassert(editor->filterResonanceSlider[whichFilter] != nullptr); // filter.resonance_1 or filter.resonance_2 is not assigned in SurgeGuiEditor
+        editor->filterCutoffSlider[whichFilter]->asControlValueInterface()->setValue(f);
+        editor->filterCutoffSlider[whichFilter]->setQuantitizedDisplayValue(f);
+        editor->filterCutoffSlider[whichFilter]->asJuceComponent()->repaint();
+        editor->filterResonanceSlider[whichFilter]->asControlValueInterface()->setValue(resonance);
+        editor->filterResonanceSlider[whichFilter]->setQuantitizedDisplayValue(resonance);
+        editor->filterResonanceSlider[whichFilter]->asJuceComponent()->repaint();
+        repushData();
+    }
+}
+void FilterAnalysis::mouseDown(const juce::MouseEvent &event) {
```

Comment:
I'm assuming this is here because you need to update the sliders when the mouse first goes down in addition to when it drags. Could you add a comment here saying that? I don't want to come back a year later and not be sure what's going on :-)

Reply:
Done